compute_exclusivity.py

```diff
@@ -54,25 +83,46 @@ def run( args ):
     # Create a list of sets to test
     sets = list( frozenset(t) for t in combinations(genes, args.gene_set_size) )
     num_sets = len(sets)
-    
+
+    # Load the weights (if necessary)
+    test = nameToTest[args.test]
+    if test == WEIGHTED:
+        # Master matrix of all weights
+        P = np.zeros((num_genes, num_patients))
+        patient_index = 0
+        for i, weights_file in enumerate(args.weights_files):
+            # Load the weights matrix for this cancer type and update the entires appropriately
+            type_P          = np.load(weights_file)
+            ty_num_patients = np.shape(type_P)[1]
+            ty_genes        = set(typeToGenes[i]) & genes
+            ty_gene_indices = [ typeToGeneIndex[i][g] for g in ty_genes ]
+            gene_indices    = [ geneToIndex[g] for g in ty_genes ]
+            P[gene_indices, patient_index:patient_index + ty_num_patients] = type_P[ ty_gene_indices ]
+            patient_index  += ty_num_patients
+            
+        # Set any zero entries to the minimum (pseudocount)
+        P[P == 0] = np.min(P[P > 0])
+        geneToP = dict( (g, P[geneToIndex[g]]) for g in genes )
+    else:
+        geneToP = None
+        
     # Test each set
     print '* Testing {} sets...'.format(num_sets)
-    test   = nameToTest[args.test]
     if test == PERMUTATIONAL:
-        # Restrict the list of files we're testing
-        permuted_files = args.permuted_matrix_files[:args.num_permutations]
+        # Group and restrict the list of files we're testing
+        permuted_directory_files = []
+        for permuted_matrix_dir in args.permuted_matrix_directories:
+            files = os.listdir(permuted_matrix_dir)
+            permuted_matrices = [ '{}/{}'.format(permuted_matrix_dir, f) for f in files ]
+            permuted_directory_files.append( permuted_matrices[:args.num_permutations] )
+        assert( len(files) == args.num_permutations for files in permuted_directory_files )
+        
+        permuted_files = zip(*permuted_directory_files)
         print '\tUsing {} permuted matrix files'.format(len(permuted_files))
 
         # Run the permutational
         setToPval, setToRuntime, setToFDR, setToObs = permutational_test( sets, geneToCases, num_patients, permuted_files, args.num_cores, args.verbose )
     else:
```

Comment:
Let's replace `else:` with `elif test == UNWEIGHTED:` and add `else: Raise NotImplementedError('...')`.